src/background/hypothesis-chrome-extension.js

```diff
@@ -178,14 +183,23 @@ export default function HypothesisChromeExtension({
   // Two important events are when the tab's `status` changes to `loading`
   // when the user begins a new navigation and when the tab's status changes
   // to `complete` after the user completes a navigation
-  function onTabUpdated(tabId, changeInfo, tab) {
-    if (changeInfo.status === TAB_STATUS_LOADING) {
-      resetTabState(tabId, tab.url);
-      const query = directLinkQuery(tab.url);
+  /**
+   *
+   * @param {number} tabId
+   * @param {chrome.tabs.TabChangeInfo} changeInfo
+   * @param {chrome.tabs.Tab} tab
+   */
+  function onTabUpdated(tabId, { status }, tab) {
+    if (status === 'loading' && !isLoading.has(tabId)) {
+      isLoading.add(tabId);
+      const url = /** @type {string} - property included because manifest has the "tabs" permission*/ (tab.url);
+      resetTabState(tabId, url);
```

Comment:
If the URL changes after the `loading` status is emitted for a tab for the first time and before a `complete` status is emitted then `resetTabState` won't be called. Can this happen if redirects occur or if the user navigates to a different URL before loading has complete for example?

Reply:
This solution works for:
* server-side redirects
* client-side redirects
* click on the 'Stop loading this page` button

However, it doesn't work if the user navigates before the loading of the document has completed. I was under the assumption that the status `complete` was fired before the `loading` of the new URL in this cases. My assumption was wrong. Thanks for suggesting this scenario.

I have a feeling we are using the wrong API for this and that we should probably be using the [webNavigation API](https://developer.chrome.com/extensions/webNavigation).